### ISSUE TITLE
Clicking ID field in Table view shows error

### ISSUE THREAD

**Issue by mksplg** [metabase/metabase#13263]
**Describe the bug**
When clicking on a primary key field in the table view of a query, the detail view of the record should be opened. Instead an error message "There was a problem with your question" with the postgres error `ERROR: operator does not exist: bigint = character varying Hint: No operator matches the given name and argument type(s). You might need to add explicit type casts. Position: 1323` is shown.

Showing a detail page works by adding a filter to the custom query with the given ID manually.

**Logs**
[metabase logs.txt](https://github.com/metabase/metabase/files/5238706/metabase.logs.txt)

**To Reproduce**
1. Create a new custom question
2. Select a table with a DB id, the id should be shown as a link to the details page for the record
3. Click the link

**Expected behavior**
Clicking on a primary key should open the details page and show the selected record.

**Screenshots**
![grafik](https://user-images.githubusercontent.com/1591254/93464229-229cef00-f8e9-11ea-992f-fc36631cf390.png)
![grafik](https://user-images.githubusercontent.com/1591254/93464261-3183a180-f8e9-11ea-918e-05514c5430c9.png)

**Information about your Metabase Installation:**

```json
{
  "browser-info": {
    "language": "de",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:81.0) Gecko/20100101 Firefox/81.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.8+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.8",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.8+10",
    "os.name": "Linux",
    "os.version": "4.4.0-165-generic",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "mysql",
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.4 (Debian 12.4-1.pgdg100+1)"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-09-15",
      "tag": "v0.36.6",
      "branch": "release-0.36.x",
      "hash": "cb258fb"
    },
    "settings": {
      "report-timezone": "Europe/Berlin"
    }
  }
}
```

**Severity**
It is this annoying, blocking an upgrade.
Predefined questions/dashboards still work but it's not possible to browse through data anymore.

**Additional context**
The `ID` columns are set as "Entity Key". The problem is for all tables.

**Comment by flamber** [comment]
The same as #13249, but it's any PK column size on Postgres

**Comment by mksplg** [comment]
Sorry, I didn't see the other issue.

Note that the primary key fields are also `BigInt` in my case. See the attached logs.

**Comment by saarwasserman** [comment]
**Do you know what is the latest version that doesn't cause this issue?**

I use Integer type for all primary keys and foreign keys but the sql editor shows it makes comparison to a string 'id' instead of id
db: postgres.

**Comment by flamber** [comment]
@saarwasserman Downgrade to 0.36.4

**Comment by camsaul** [comment]
Fixed by #13443

**Comment by mksplg** [comment]
I can confirm that the issue is fixed for me in 0.36.8.

Thank you.
